imap_processing/hit/l0/data_classes/housekeeping.py

```diff
@@ -0,0 +1,219 @@
+    EBOX_P5D7VA : int
+        +5.7VA Ebox (analog)
+    EBOX_M5D7VA : int
+        -5.7VA Ebox (analog)
+    REF_P5V : int
+        +5Vref
+    L1AB_BIAS : int
+        L1A/B Bias
+    L2AB_BIAS : int
+        L2A/B Bias
+    L34A_BIAS : int
+        L3/4A Bias
+    L34B_BIAS : int
+        L3/4B Bias
+    EBOX_P2D0VD : int
+        +2.0VD Ebox (digital)
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
+        Uses the CCSDS packet, version of the software, and
+        the name of the packet file to parse and store information about
+        the Houskeeping packet data.
+    _parse_leak():
+        Parse each current leakage field and put into an array.
+
+    """
+    SHCOARSE: int
+    MODE: int
+    FSW_VERSION_A: int
+    FSW_VERSION_B: int
+    FSW_VERSION_C: int
+    NUM_GOOD_CMDS: int
+    LAST_GOOD_CMD: int
+    LAST_GOOD_SEQ_NUM: int
+    NUM_BAD_CMDS: int
+    LAST_BAD_CMD: int
+    LAST_BAD_SEQ_NUM: int
+    FEE_RUNNING: int
+    MRAM_DISABLED: int
+    ENABLE_50KHZ: int
+    ENABLE_HVPS: int
+    TABLE_STATUS: int
+    HEATER_CONTROL: int
+    ADC_MODE: int
+    DYN_THRESH_LVL: int
+    NUM_EVNT_LAST_HK: int
+    NUM_ERRORS: int
+    LAST_ERROR_NUM: int
+    CODE_CHECKSUM: int
+    SPIN_PERIOD_SHORT: int
+    SPIN_PERIOD_LONG: int
+    LEAK_I_RAW: str
+    LEAK_I: np.ndarray
+    PHASIC_STAT: int
+    ACTIVE_HEATER: int
+    HEATER_ON: int
+    TEST_PULSER_ON: int
+    DAC0_ENABLE: int
+    DAC1_ENABLE: int
+    PREAMP_L234A: int
+    PREAMP_L1A: int
+    PREAMP_L1B: int
+    PREAMP_L234B: int
+    TEMP0: int
+    TEMP1: int
+    TEMP2: int
+    TEMP3: int
+    ANALOG_TEMP: int
+    HVPS_TEMP: int
+    IDPU_TEMP: int
+    LVPS_TEMP: int
+    EBOX_3D4VD: int
+    EBOX_5D1VD: int
+    EBOX_P12VA: int
+    EBOX_M12VA: int
+    EBOX_P5D7VA: int
+    EBOX_M5D7VA: int
+    REF_P5V: int
+    L1AB_BIAS: int
+    L2AB_BIAS: int
+    L34A_BIAS: int
+    L34B_BIAS: int
+    EBOX_P2D0VD: int
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        """Intialization method for Housekeeping Data class."""
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
+        self.parse_data(packet)
+        self._parse_leak()
+    
+    def _parse_leak(self):
+        """Parse each current leakage field and put into an array."""
+        leak_list = list()
+        leak_bits = bitstring.Bits(bin=self.LEAK_I_RAW)
+        # Each Leak field is 10 bits long
+        leak_bit_length = 10
+        # There are 64 leak fields
+        num_leak_fields = 64
+        # The leak fields appear in the packet in ascending order, so to append
+        # the leak fields in the correct order, the binary will be parsed 
+        # from right to left.
+        for leak_idx in range(leak_bit_length * num_leak_fields, 0, -leak_bit_length):
+            leak_list.append(leak_bits[leak_idx - leak_bit_length:leak_idx].uint)
+        self.LEAK_I = np.array(leak_list)
```

Comment:
You can initialize the shape of the array right away and then fill in the values and avoid bitstring as well I think.
```suggestion
        # Each Leak field is 10 bits long
        leak_bit_length = 10
        # There are 64 leak fields
        num_leak_fields = 64
        self.LEAK_I = np.empty(num_leak_fields, dtype=np.uint16)
        # The leak fields appear in the packet in ascending order, so to append
        # the leak fields in the correct order, the binary will be parsed 
        # from right to left.
        for leak_idx in range(leak_bit_length * num_leak_fields, 0, -leak_bit_length):
            self.LEAK_I[i] = int(self.LEAK_I_RAW[leak_idx - leak_bit_length:leak_idx], 2)
```

Reply:
They're just stored in reverse. There's 64 leak fields in the packet and they're ordered as leak_64, leak_63, .... so I just need to reverse the field order when I make the array

Reply:
Are we trying to move away from the `bitstring` library entirely?

Reply:
IMO we should. Let bitstring be a detail for space_packet_parser, but not bring it into imap_processing if we can avoid it. I found it to be really slow. If we need to do complicated bit manipulations then I don't see much of an issue, but everything I've seen so far is not _too_ bad to handle ourselves. Note that bitstring has already made this a string for you, so you're sort of going back to bitstring then rather than using the original bits.